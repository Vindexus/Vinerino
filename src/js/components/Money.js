import React from 'react'

export default class Money extends React.Component {
  render () {
    const totalCents = this.props.amount
    let display = 0

    if(!totalCents) {
      return <span class="money money-zero">$0</span>
    }

    const dollars = Math.floor(totalCents / 100)
    const cents = totalCents - (dollars * 100)

    if(dollars >= 1000) {
      let ks = parseFloat((dollars / 1000).toFixed(2))
      display = ks + 'k'
    }
    else if(dollars >= 10) {
      display = parseFloat((dollars+cents).toFixed(2))
    }
    else {
      display = dollars.toString() + '.' + (cents < 10 ? '0' + cents.toString() : cents).toString()
    }

    return <span class="money">${display}</span>
  }
}