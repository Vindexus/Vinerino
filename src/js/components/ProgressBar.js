import React from 'react'

import Money from './Money'

export default class ProgressBar extends React.Component {
  render () {
    const { lastStretch, currentAmount, progress } = this.props.matcherino

    if(!lastStretch) {
      return null
    }

    const innerBarStyle = {
      width: (progress * 100).toFixed(2) + '%'
    }

    return (
      <div class="progress-bar-container">
        <div class="progress-bar-outer">
          <div class="progress-bar-inner" style={innerBarStyle}></div>
        </div>
        <div class="progress-bar-numbers"><Money amount={currentAmount} />/<Money amount={lastStretch.goal} /></div>
      </div>
    )
  }
}