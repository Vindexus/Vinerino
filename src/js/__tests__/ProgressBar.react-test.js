// Link.react-test.js
import React from 'react';
import ProgressBar from '../Components/ProgressBar';
import Money from '../Components/Money';
import renderer from 'react-test-renderer';

import {shallow} from 'enzyme';

const matcherinoProp = {
  lastStretch: {
    goal: 30000 //Last goal is at $300
  },
  currentAmount: 20000, //Currently at $200
  progress: 2/3
}

test('Component: ProgressBar', () => {
  const component = renderer.create(
    <ProgressBar matcherino={matcherinoProp} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  let pb = shallow(
    <ProgressBar matcherino={matcherinoProp} />
  );

  const props = pb.props()
  const innerBarProps = props.children[0].props.children.props
  expect(innerBarProps.style.width).toEqual('66.67%')
  expect(pb.find('Money')).toHaveLength(2);
})