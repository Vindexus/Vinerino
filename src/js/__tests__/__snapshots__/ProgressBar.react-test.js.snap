// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProgressBar 1`] = `
<div
  className="progress-bar-container"
>
  <div
    className="progress-bar-outer"
  >
    <div
      className="progress-bar-inner"
      style={
        Object {
          "width": "66.67%",
        }
      }
    />
  </div>
  <div
    className="progress-bar-numbers"
  >
    <span
      className="money"
    >
      $
      200
    </span>
    /
    <span
      className="money"
    >
      $
      300
    </span>
  </div>
</div>
`;
